### PR TITLE
71646 - create catalog.031 file and add to aio exchange set

### PR DESCRIPTION
**Pull request by harshda15125**
- Added code changes
- Added unit test cases
- Added functional tests

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 1]
```diff
@@ -716,7 +716,19 @@ public enum EventIds
...
+        /// </summary>
+        CatalogFileForAioIsNotCreated = 805178,
+        /// <summary>
+        /// 805179 - Request for creating catalog.031 file in aio exchange set is started.
+        /// </summary>
+        CreateCatalogFileForAioRequestStart = 805179,
+        /// <summary>
+        /// 805180 - Request for creating catalog.031 file in aio exchange set is completed.
+        /// </summary>
+        CreateCatalogFileForAioRequestCompleted = 805180,
```
Comment (shailen14599) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs:731]:
```suggestion
      
```
I think, as we have added temporary method for catalog so eventIds added with new method are also temporary. Please remove if this consumption is correct and use existing eventIds.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs [hunk 2]
```diff
@@ -36,6 +36,7 @@ public class FulfilmentAncillaryFilesTest
         public string fakeExchangeSetInfoPath = @"C:\\HOME";
         public string fulfilmentExceptionMessage = "There has been a problem in creating your exchange set, so we are unable to fulfil your request at this time. Please contact UKHO Customer Services quoting error code : {0} and correlation ID : {1}";
         private readonly DateTime fakeScsRequestDateTime = DateTime.UtcNow;
+
```
Comment (shailen14599) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs:39]:
```suggestion
```
Remove this additional blank line

**Review by shailen14599 — CHANGES_REQUESTED** [review]
Minor changes requested

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs (hunk 1 above)
Comment (harshda15125) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs (hunk 2 above)
Comment (harshda15125) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentAncillaryFilesTest.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs [hunk 3]
```diff
@@ -88,6 +90,8 @@ public void Setup()
             fakeproductDataValidator = A.Fake<IProductDataValidator>();
             fakeAioConfiguration = A.Fake<IOptions<AioConfiguration>>();
 
+
+
```
[the remark this answers is not in the document]
Comment (harshda15125) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs]: Resolved

**Review by shailen14599 — APPROVED** [review]
LGTM

**Review by DanOwenUKHO — APPROVED** [review]
LGTM, noted the temporary methods to be replaced in future PR

**Pull request merged**